src/hotspot/share/gc/shenandoah/shenandoahGlobalGeneration.cpp

```diff
@@ -58,6 +58,14 @@ size_t ShenandoahGlobalGeneration::available() const {
   return ShenandoahHeap::heap()->free_set()->available();
 }
 
+size_t ShenandoahGlobalGeneration::soft_available() const {
+  size_t available = this->available();
+
+  // Make sure the code below treats available without the soft tail.
+  size_t soft_tail = max_capacity() - soft_max_capacity();
```

Comment:
Would it be useful to assert here that `max_capacity() >= soft_max_capacity()` ?

Reply:
Yes, better than computing an underflowed valued.